libass/ass_font.c

```diff
@@ -95,8 +95,16 @@ uint32_t ass_font_index_magic(FT_Face face, uint32_t symbol)
     }
 }
 
-static void buggy_font_workaround(FT_Face face)
+static void set_font_metrics(FT_Face face)
 {
+    // Mimicking GDI's behavior for asc/desc/height
+    TT_OS2 *os2 = FT_Get_Sfnt_Table(face, ft_sfnt_os2);
```

Comment:
There is the same os2 table later requested in this function. At least second request should be deleted, but even branch conditions have some overlap, maybe better to merge that too.